app/graph.py

```diff
@@ -470,23 +493,112 @@ def update(self):
+                        "primary_iface": "",
+                        "ns": i.metadata.namespace,
+                        "labels": i.metadata.labels if i.metadata.labels is not None else {},
+                        "other_ifaces": {},
+                        "annotations": i.metadata.annotations if i.metadata.annotations is not None else {},
                     }
 
-                self.topology['nodes'][i.spec.node_name]['pods'][i.metadata.name] = {
-                    "ip": i.status.pod_ip,
-                    "ns": i.metadata.namespace,
-                    "labels": i.metadata.labels if i.metadata.labels is not None else {}
-                    }
-                    
+                    annotations = i.metadata.annotations
+                    if annotations is not None:
+                        for key, annotation in annotations.items():
+                            if key == "k8s.v1.cni.cncf.io/network-status":
+                                items_list = json.loads(annotation)
+                                pod_ip = i.status.pod_ip
+                                for val in items_list:
+                                    if val["ips"][0] != pod_ip:
+                                        iface_name = str(val["name"].split('/')[-1])
+                                        pod_iface = str(val["interface"])
+                                        pods[pod_name]['other_ifaces'][iface_name] = pod_iface
+                                    else:
+                                        pods[pod_name]['primary_iface'] = str(val["interface"])
+            except Exception as e:
+                logger.error(f"Error processing pod: {i.metadata.name}. Error: {str(e)}")
+
+        cr = self.custom_obj.list_namespaced_custom_object(group="aci.fabricattachment", version="v1", namespace="aci-containers-system", plural="nodefabricnetworkattachments")
+
+        if cr.get("items"):
+            self.nfna_cr = True
+            for nodeName in self.topology['nodes']:
+                try:
+                    for i in cr.get("items"):
+                        aciTopology = i["spec"].get("aciTopology")
+                        if aciTopology is not None and i["spec"]["nodeName"] == nodeName:
+                            for iface_name, link in i["spec"]["aciTopology"].items():
+                                if "sriov" in i["metadata"]["name"]:
+                                    iface_name = "PF-" + iface_name
+                                fabricLinks = link.get("fabricLink", [])
+                                for fabricLink in fabricLinks:
+                                    fabricLinkSplit = fabricLink.split("/")
+                                    switch_name = fabricLinkSplit[2].replace("node", "leaf")
+                                    switch_interface = fabricLink[fabricLink.rfind('[') + 1: fabricLink.rfind(']')]
+                                    if "sriov" in i["metadata"]["name"]:
```

Comment:
we need to come up with a more robust solution that how we want to check for the type. The name might not be enough. Will discuss in the group want we want to select a type - a name, or an annotation